tests/testthat/test-process_directory.R

```diff
@@ -0,0 +1,88 @@
+# suppress log messages during testing.
+old_threshold <- logger::log_threshold()
+withr::defer(logger::log_threshold(old_threshold))
+logger::log_threshold("FATAL")
+
+empty_groups <- data.frame()
+simple_groups <- tibble::tribble(
+  ~investor_name, ~portfolio_name,
+  "Simple Investor", "Simple Portfolio"
+)
+
+test_that("Processing a directory with a single file works.", {
+  test_file <- testthat::test_path(
+    "testdata", "portfolios", "simple.csv"
+  )
+  filehash <- digest::digest(
+    object = test_file,
+    file = TRUE,
+    algo = "md5"
+  )
+  input_dir <- tempfile("input")
+  dir.create(input_dir)
+  withr::defer(unlink(input_dir))
+  output_dir <- tempfile("output")
```

Comment:
```suggestion
  input_dir <- tempdir("input")
  dir.create(input_dir)
  withr::defer(unlink(input_dir))
  output_dir <- tempdir("output")
```
should these be using `tempdir()`?

Reply:
`tempdir` does not accept a naming argument. In fact, within a single session of R (`pid`), it always returns the same value.

However, creating a directory with `tempfile` as done here allow for differentiating names and `tempfile()` by deault does not include an extension, which is nice for creating directories.